### DISCUSSION TITLE
Mobx stores values are getting leaked on server

### DISCUSSION THREAD

**Discussion by thebedroomprogrammer** [mobxjs/mobx#2908] [General]
I am using mobx for state management. My approach is to generate the state on server and inject it in the React components using React context. The error that I am facing is that whenever I render a component on the server that access the state the server holds onto the store in the memory and for each request a new store is created which remains in the memory causing memory leakage. I am not sure If my approach is wrong. I am pasting the code which is causing the issue. 

Also, I am using useStaticRendering(true) on the server

    "mobx": "^5.13.0",
    "mobx-react": "^6.1.1",

  ```
const FooterContainer: React.FC<{}> = () => {
    const [isOpen, toggleFooter] = React.useState(TOGGLE_STATE.INACTIVE);
    const { ui, env, user } = React.useContext(StoreContext);
    const controller = new FooterController();
    const location = useLocation();
  
    const [isLoading, setLanguage] = useLazyEpisode(true, controller.setLanguage);
  
    const changeLang = async (lang: LANG_SET) => {
      const fromLang = env.lang;
      const toLang = lang;  
      doSomething(fromLang,toLang)
    };
  
    return (
      <Observer>
        {() => {
          if (ui.footerEnabled) {
            return (
              <FooterView
                changeLang={changeLang}
                onlyLang={ui.currentRoute?.footerConfig?.onlyLang}
                lang={env.lang}
                gzpLocale={env.gzpLocale}
                isOpen={isOpen}
                toggleFooter={() => {
                  toggleFooter(isOpen === TOGGLE_STATE.INACTIVE ? TOGGLE_STATE.ACTIVE : TOGGLE_STATE.INACTIVE);
                }}
              />
            );
          } else {
            return <></>;
          }
        }}
      </Observer>
    );
  };
```

The memory leakage problem goes away when I remove the changeLang function

![image](https://user-images.githubusercontent.com/20200740/115729565-245cf280-a3a3-11eb-8598-21c4bdd79287.png)

In the image above you can see that the Environment store is created 4 times in the memory for 4 request causing the memory leak as it is not garbage collected. 

Thanks.

**Comment by mweststrate** [comment]
Hi, would you mind filing a bug report instead by filling out the template and providing a minimal reproduction? Screenshots and some snippets of code aren't actionable for us and so far there isn't anything in here that points to MobX causing the problem.
